operator/cmd/main.go

```diff
@@ -0,0 +1,167 @@
+	utilruntime.Must(ocpnetworkv1alpha1.Install(scheme))
+
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var (
+		metricsAddr              string
+		enableLeaderElection     bool
+		probeAddr                string
+		secureMetrics            bool
+		enableHTTP2              bool
+		coreDNSNamespace         string
+		coreDNSServieName        string
+		dnsNameResolverNamespace string
+	)
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&secureMetrics, "metrics-secure", false,
+		"If set the metrics endpoint is served securely")
+	flag.BoolVar(&enableHTTP2, "enable-http2", false,
+		"If set, HTTP/2 will be enabled for the metrics and webhook servers")
+	flag.StringVar(&coreDNSNamespace, "coredns-namespace", "kube-system", "The namespace of the CoreDNS resources.")
+	flag.StringVar(&coreDNSServieName, "coredns-service-name", "kube-dns", "The name of the CoreDNS service.")
+	flag.StringVar(&dnsNameResolverNamespace, "dns-name-resolver-namespace", "ovn-kubernetes",
+		"The namespace to watch for the DNSNameResolver objects.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	// if the enable-http2 flag is false (the default), http/2 should be disabled
+	// due to its vulnerabilities. More specifically, disabling http/2 will
+	// prevent from being vulnerable to the HTTP/2 Stream Cancelation and
+	// Rapid Reset CVEs. For more information see:
+	// - https://github.com/advisories/GHSA-qppj-fm5r-hxr3
+	// - https://github.com/advisories/GHSA-4374-p667-p6c8
+	disableHTTP2 := func(c *tls.Config) {
+		setupLog.Info("disabling http/2")
+		c.NextProtos = []string{"http/1.1"}
+	}
+
+	tlsOpts := []func(*tls.Config){}
+	if !enableHTTP2 {
+		tlsOpts = append(tlsOpts, disableHTTP2)
+	}
+
+	webhookServer := webhook.NewServer(webhook.Options{
+		TLSOpts: tlsOpts,
+	})
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress:   metricsAddr,
+			SecureServing: secureMetrics,
+			TLSOpts:       tlsOpts,
+		},
+		WebhookServer:          webhookServer,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "3a3a07d4.openshift.io",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// Set up the DNSNameResolver controller.  This controller is unmanaged by
+	// the manager; the dnsnameresolvercrd controller starts it and the
+	// caches after it creates the DNSNameResolver CRD.
```

Comment:
The reason why the controller is unmanaged is so that it doesn't get automatically started when the operator starts; we only want the controller to start if we need it.  It would be good for the comment to explain this.

Reply:
As discussed in [this thread](https://redhat-internal.slack.com/archives/C04VCUBSU4U/p1713964224175029), let's follow up on this one in a separate PR.